renku-model/src/test/scala/io/renku/graph/model/entities/PlanSpec.scala

```diff
@@ -186,45 +187,15 @@ class PlanSpec
       }
     }
 
-    "pick up the correct decoder" in {
-      val cp = compositePlanGen().generateOne
-        .to[entities.CompositePlan]
-
+    "decode any Plan entity subtypes" in {
+      val cp = compositePlanGen().generateOne.to[entities.CompositePlan]
       val sp = plans.generateOne.to[entities.StepPlan]
 
-      val spDecoded = flattenedJsonLD(sp).cursor.as[List[entities.Plan]]
-      val cpDecoded = flattenedJsonLD(cp).cursor.as[List[entities.Plan]]
-
-      spDecoded shouldBe Right(List(sp))
-      cpDecoded shouldBe Right(List(cp))
-    }
-
-    "fail decode if not strictly matching entity types (composite plan)" in {
-      val cp = compositePlanGen().generateOne
-        .to[entities.CompositePlan]
-
-      val newJson =
-        JsonLDTools
-          .view(cp)
-          .selectByTypes(entities.CompositePlan.Ontology.entityTypes)
-          .addType(renku / "WorkflowCompositePlan")
-          .value
-
-      newJson.cursor.as[List[entities.CompositePlan]] shouldBe Right(Nil)
-    }
-
-    "fail decode if not strictly matching entity types (step plan)" in {
-      val sp = stepPlanEntities().generateOne
-        .to[entities.StepPlan]
+      val jsonLD = flattenedJsonLDFrom(cp.asJsonLD, sp.asJsonLD)
 
-      val newJson =
-        JsonLDTools
-          .view(sp)
-          .selectByTypes(entities.StepPlan.entityTypes)
-          .addType(renku / "WorkflowPlan")
-          .value
+      val decoded = jsonLD.cursor.as[List[entities.Plan]]
 
-      newJson.cursor.as[List[entities.StepPlan]] shouldBe Right(Nil)
+      decoded.fold(err => fail(err.message), identity) should contain theSameElementsAs List(sp, cp)
```

Comment:
A test for the case we have an additional type in the jsonld would also be useful, no? like the removed `fail decode if…` tests (that didn't really fail, but returned an empty list) but on the `Plan.decoder` level (not on the subtypes) - iiuc that was the problem yesterday?

Reply:
Yes, we could have one. Let me think if I can compile something like that.